### ISSUE TITLE
Broken `AxiosHeaders` typings

### ISSUE THREAD

**Issue by nfantone** [axios/axios#5415]
### Describe the bug

`AxiosHeaders` types makes it so trying to call methods on the instance raises a `typescript` compilation error.

<img width="1142" alt="image" src="https://user-images.githubusercontent.com/476069/209943677-fa7775e7-0f23-4b86-88ae-ad28df8c0533.png">

`AxiosRequestConfig.headers` can be either `RawAxiosRequestHeaders` or `AxiosHeaders`. `RawAxiosRequestHeaders`, being an alias of `Record<string, AxiosHeaderValue>` causes methods to not be callable.

### To Reproduce

Invoke any method on `AxiosRequestConfig.headers`.

### Code snippet

```js
const api = axios.create({ ... })

api.interceptors.request.use(req => {
  // Trying to access any method on `req.headers` will result in
  // a "This expression is not callable" TS error
  req.headers.set('foo', 'bar')
})
```

### Expected behavior

Should be able to call any method declared in `AxiosHeaders`.

### Environment

* `axios`: `1.2.2`
* `node`: `18.12.1`
* `typescript`: `4.9.4`
* macOS Ventura `13.1`

**Comment by nfantone** [comment]
In addition to the above, since `AxiosRequestConfig.headers` is an optional property, it's not straightforward to set headers using regular object access notation — i.e.: `config.headers['foo']`.

<img width="1175" alt="image" src="https://user-images.githubusercontent.com/476069/209948746-d195fffd-4271-4984-98e1-91920b15119b.png">

Is the expectation to coerce the type of `.headers` before accessing it?

**Comment by karlhorky** [comment]
Seems like this is also similar to #5416, which may also be fixed by #5420
